src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHash3.cs

```diff
@@ -0,0 +1,884 @@
+                        if (length >= 16 * 12)
+                        {
+                            accumulators += Mix16Bytes(source + (16 * 11), 0x9035E08E2443F774, 0x52283C4C263A81E6, seed); // DefaultSecret[51], DefaultSecret[59]
+                            if (length >= 16 * 13)
+                            {
+                                accumulators += Mix16Bytes(source + (16 * 12), 0x65D088CB00C391BB, 0x486471A32E531B8B, seed); // DefaultSecret[67], DefaultSecret[75]
+                                if (length >= 16 * 14)
+                                {
+                                    accumulators += Mix16Bytes(source + (16 * 13), 0xEF19384EF90DA297, 0x76FAA8D8ACDEA946, seed); // DefaultSecret[83], DefaultSecret[91]
+                                    if (length >= 16 * 15)
+                                    {
+                                        accumulators += Mix16Bytes(source + (16 * 14), 0xBBDCF93F349CE33F, 0xE0518A1D4F0BC7C7, seed); // DefaultSecret[99], DefaultSecret[107]
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Handle the last 16 bytes.
+            accumulators += Mix16Bytes(source + length - 16, 0x7378D9C97E9FC831, 0xEBD33483ACC5EA64, seed); // DefaultSecret[119], DefaultSecret[127]
+            return Avalanche(accumulators);
+        }
+
+        private static ulong HashLengthOver240(byte* source, uint length, ulong seed)
+        {
+            Debug.Assert(length > 240);
+
+            fixed (byte* defaultSecret = &MemoryMarshal.GetReference(DefaultSecret))
+            {
+                byte* secret = defaultSecret;
+                if (seed != 0)
+                {
+                    byte* customSecret = stackalloc byte[SecretLengthBytes];
+                    DeriveSecretFromSeed(customSecret, seed);
+                    secret = customSecret;
+                }
+
+                ulong* accumulators = stackalloc ulong[AccumulatorCount];
+                InitializeAccumulators(accumulators);
+
+                const int StripesPerBlock = (SecretLengthBytes - StripeLengthBytes) / SecretConsumeRateBytes;
+                const int BlockLen = StripeLengthBytes * StripesPerBlock;
+                int blocksNum = (int)((length - 1) / BlockLen);
+
+                uint offset = 0;
+                for (int i = 0; i < blocksNum; i++)
+                {
+                    Accumulate(accumulators, source + offset, secret, StripesPerBlock);
+                    ScrambleAccumulators(accumulators, secret + (SecretLengthBytes - StripeLengthBytes));
+                    offset += BlockLen;
+                }
+
+                int stripesNumber = (int)((length - 1 - (BlockLen * blocksNum)) / StripeLengthBytes);
```

Comment:
```suggestion
                int stripesNumber = (int)((length - 1 - offset) / StripeLengthBytes);
```

`BlockLen * blocksNum` is already known (`offset`)